broadinstitute/pearl
# [JN-828] Track viewed language in form responses

#### DESCRIPTION

Updates the Answer table to store the code of the language that was being viewed when the participant submitted their responses. Also displays the language name in the form response admin view:

![Screenshot 2024-02-13 at 12 55 30 PM](https://github.com/broadinstitute/juniper/assets/7257391/eadc92cf-782d-4d53-bebc-6ab5da551c56)


#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

Test via Populate:

* Reboot AdminApi + Admin UI
* Repopulate demo: `./scripts/populate_portal.sh demo`
* View the pre-reg, consent, and The Basics responses for Jonas Salk. Verify that you see a few "Answered in ..." displaying next to the questionStableId. Note that not all The Basics answers come pre-populated with language data: https://localhost:3000/demo/studies/heartdemo/env/sandbox/participants/HDSALK/consents/hd_hd_consent
* Verify no regressions in this view for a single-language study (i.e. OurHealth): https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/participants/OHSALK/surveys/oh_oh_basicInfo

Test with new participant:

* Reboot ParticipantApi + Participant UI (I suggest booting the UI with `REACT_APP_UNAUTHED_LOGIN=true HTTPS=true npm -w ui-participant start` so you can bypass B2C)
* Register as a new user in the Heart Demo study: https://sandbox.demo.localhost:3001
* Go to the Admin UI and view the form response data for your new participant
* Verify that you can see the language used when you filled out the form
